cs_upload.py

```diff
@@ -203,14 +203,27 @@ def upload_to_s3(self, form_data: str, trace_file: str) -> None:
         fields["success_action_status"] = str(fields["success_action_status"])
         fields["x-amz-meta-filename"] = basename(trace_file)
 
+        (_, extension) = splitext(trace_file)
+        if extension == ".gz":
+            mime_type = "application/gzip"
+        elif extension == ".cst":
+            mime_type = ""
+        else:
+            print("Trace file extension must be either .gz or .cst")
+            exit(1)
+
         query = ["curl"]
         if self.ca_cert:
             query += ["--cacert", self.ca_cert]
         for (key, value) in fields.items():
             query += ["--form", f"{key}={value}"]
         query += [
             "--form",
-            f"file=@{trace_file};type=application/gzip",
+            f"Content-Type={mime_type}",
+            "--form",
+            f"file=@{trace_file};type={mime_type}"
```

Comment:
Do you know why the `type` attribute is used? I've never seen that before, either in our documentation or our tools.

Reply:
It's mentioned in the `curl` documentation, both in the manpage and the tutorial. (See https://curl.se/docs/manpage.html and https://curl.se/docs/manual.html.) I'm not sure I understand how this relates to the `--form Content-Type=...` form element. I was putting it in to be safe, in case `curl` needed it, but I just tested without it and apparently it's not necessary, so I've dropped it. Simpler is better, all other things being equal.

Reply:
Ah right, so they have indeed different purposes. For example:

```bash
curl --form 'Content-Type=type/b' --form 'foo-field=@foo;type=type/a' http://httpbin.org/post
```

leads to the following request:

```http
POST /post HTTP/1.1
Host: httpbin.org
User-Agent: curl/7.83.1
Accept: */*
Content-Length: 304
Content-Type: multipart/form-data; boundary=------------------------440141f7184eccbc

--------------------------440141f7184eccbc
Content-Disposition: form-data; name="Content-Type"

type/b
--------------------------440141f7184eccbc
Content-Disposition: form-data; name="foo-field"; filename="foo"
Content-Type: type/a

content-of-foo

--------------------------440141f7184eccbc--
```